### ISSUE TITLE
How do you run your zipped Python packages?

### ISSUE THREAD

**Issue by kyle-figure** [oxidase/rules_poetry#49]
@oxidase Hello, I was trying out your repo but couldn't figure out how to run the .zip file after building `//:deploy_arm64`. When using the "python_zip_file" filegroup target provided by "rules_python", I am able to run their .zip files just by passing it into the Python3 interpreter.
![image](https://github.com/oxidase/rules_poetry/assets/117235991/580ae274-3e76-4209-af55-0198a12d81f3)

I am using Ubuntu 22.04 and Bazel 7.0.2.

**Comment by oxidase** [comment]
Good catch! zip files had no `__main__.py` files. With #50 `__main__.py` files are generated from `@bazel_tools//tools/python:python_bootstrap_template.txt` template. 

Heads up, it is possible to run zip files only if the packaged python interpreter can be executed on the host platform. For a MacOS aarch64 host platform
```
bazel build //:deploy_macos ;  python3 bazel-bin/deploy_macos.zip
Python interpreter /var/folders/bw/qyh03dld5fn7s1x_vm3_rmdw0000gn/T/Bazel.runfiles_2w1ojnrz/rules_python~0.31.0~python~python_3_12_aarch64-apple-darwin/bin/python3
```
works and
```
bazel build //:deploy_x86_64 ; python3 bazel-bin/deploy_x86_64.zip

Traceback (most recent call last):
  File "/var/folders/bw/qyh03dld5fn7s1x_vm3_rmdw0000gn/T/Bazel.runfiles_m0vkerdg/_main/app", line 573, in <module>
    Main()
  File "/var/folders/bw/qyh03dld5fn7s1x_vm3_rmdw0000gn/T/Bazel.runfiles_m0vkerdg/_main/app", line 558, in Main
    ExecuteFile(
  File "/var/folders/bw/qyh03dld5fn7s1x_vm3_rmdw0000gn/T/Bazel.runfiles_m0vkerdg/_main/app", line 356, in ExecuteFile
    _RunExecv(python_program, main_filename, args, env)
  File "/var/folders/bw/qyh03dld5fn7s1x_vm3_rmdw0000gn/T/Bazel.runfiles_m0vkerdg/_main/app", line 379, in _RunExecv
    os.execv(python_program, [python_program, main_filename] + args)
OSError: [Errno 8] Exec format error: '/var/folders/bw/qyh03dld5fn7s1x_vm3_rmdw0000gn/T/Bazel.runfiles_m0vkerdg/rules_python~0.31.0~python~python_3_12_x86_64-unknown-linux-gnu/bin/python3'
```

```
bazel build //:deploy_arm64 ; python3 bazel-bin/deploy_arm64.zip

  File "/var/folders/bw/qyh03dld5fn7s1x_vm3_rmdw0000gn/T/Bazel.runfiles_dor3nu_r/_main/app", line 573, in <module>
    Main()
  File "/var/folders/bw/qyh03dld5fn7s1x_vm3_rmdw0000gn/T/Bazel.runfiles_dor3nu_r/_main/app", line 558, in Main
    ExecuteFile(
  File "/var/folders/bw/qyh03dld5fn7s1x_vm3_rmdw0000gn/T/Bazel.runfiles_dor3nu_r/_main/app", line 356, in ExecuteFile
    _RunExecv(python_program, main_filename, args, env)
  File "/var/folders/bw/qyh03dld5fn7s1x_vm3_rmdw0000gn/T/Bazel.runfiles_dor3nu_r/_main/app", line 379, in _RunExecv
    os.execv(python_program, [python_program, main_filename] + args)
OSError: [Errno 8] Exec format error: '/var/folders/bw/qyh03dld5fn7s1x_vm3_rmdw0000gn/T/Bazel.runfiles_dor3nu_r/rules_python~0.31.0~python~python_3_12_aarch64-unknown-linux-gnu/bin/python3'
```
both fail. These packaged zip files should run correctly on corresponding platforms

**Comment by kyle-figure** [comment]
Thanks for the fix! Was able to test on my MacOS arm64 host machine and Ubuntu arm64 Docker container. Running the zip file on both works as expected.
![image](https://github.com/oxidase/rules_poetry/assets/117235991/1753f0de-f883-40fd-86aa-7595dd6034c0)
![image](https://github.com/oxidase/rules_poetry/assets/117235991/ed06d9d6-4d5f-4c11-b7ff-6954e25314d7)

**Comment by oxidase** [comment]
The fix will be published in [v0.3.2](https://github.com/bazelbuild/bazel-central-registry/pull/1659)
